modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/asset/model/DLFileEntryAssetRendererFactory.java

```diff
@@ -23,7 +23,6 @@
 import com.liferay.document.library.kernel.exception.NoSuchFileEntryException;
 import com.liferay.document.library.kernel.model.DLFileEntry;
 import com.liferay.document.library.kernel.model.DLFileEntryType;
```

Comment:
The method

```
@Override
public boolean hasAddPermission(
```

is failing:

Steps to reproduce

1. Create a widget page (upgraded or from clean server)
2. Try to add a widget


**actual result:**
error on console
```

2023-01-18 10:46:03.270 ERROR [http-nio-8080-exec-10][PortletRequestDispatcherImpl:304] Unable to dispatch request: java.lang.RuntimeException: java.lang.IllegalArgumentException: Someone may be trying to circumvent the permission checker: {companyId=20097, name=com.liferay.document.library.kernel.model.DLFileEntryType, primKey=20100, scope=4}
2023-01-18 10:46:03.270 ERROR [http-nio-8080-exec-10][PortletServlet:118] Unable to process portlet com_liferay_product_navigation_control_menu_web_portlet_ProductNavigationControlMenuPortlet: org.apache.jasper.JasperException: java.lang.RuntimeException: java.lang.IllegalArgumentException: Someone may be trying to circumvent the permission checker: {companyId=20097, name=com.liferay.document.library.kernel.model.DLFileEntryType, primKey=20100, scope=4}
javax.portlet.PortletException: org.apache.jasper.JasperException: java.lang.RuntimeException: java.lang.IllegalArgumentException: Someone may be trying to circumvent the permission checker: {companyId=20097, name=com.liferay.document.library.kernel.model.DLFileEntryType, primKey=20100, scope=4}
	at com.liferay.portlet.internal.PortletRequestDispatcherImpl.dispatch(PortletRequestDispatcherImpl.java:307) ~[portal-impl.jar:?]
	at com.liferay.portlet.internal.PortletRequestDispatcherImpl.include(PortletRequestDispatcherImpl.java:123) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet.include(MVCPortlet.java:620) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet.serveResource(MVCPortlet.java:361) ~[portal-kernel.jar:?]
	at com.liferay.product.navigation.control.menu.web.internal.portlet.ProductNavigationControlMenuPortlet.serveResource(ProductNavigationControlMenuPortlet.java:71) ~[bundleFile:?]
...
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:9.0.68]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_311]
Caused by: org.apache.jasper.JasperException: java.lang.RuntimeException: java.lang.IllegalArgumentException: Someone may be trying to circumvent the permission checker: {companyId=20097, name=com.liferay.document.library.kernel.model.DLFileEntryType, primKey=20100, scope=4}
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:440) ~[jasper.jar:9.0.68]
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:473) ~[jasper.jar:9.0.68]
...
	at com.liferay.portlet.internal.PortletRequestDispatcherImpl.dispatch(PortletRequestDispatcherImpl.java:291) ~[portal-impl.jar:?]
	... 181 more
Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: Someone may be trying to circumvent the permission checker: {companyId=20097, name=com.liferay.document.library.kernel.model.DLFileEntryType, primKey=20100, scope=4}
	at com.liferay.portal.kernel.util.BaseMapWrapper.doPut(BaseMapWrapper.java:105) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.util.HashMapBuilder$HashMapWrapper.put(HashMapBuilder.java:135) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.util.HashMapBuilder.put(HashMapBuilder.java:58) ~[portal-kernel.jar:?]
	at com.liferay.product.navigation.control.menu.web.internal.display.context.AddContentPanelDisplayContext.getAddContentPanelData(AddContentPanelDisplayContext.java:94) ~[bundleFile:?]
...
	at org.eclipse.equinox.http.servlet.internal.context.DispatchTargets.doDispatch(DispatchTargets.java:120) ~[?:?]
	at org.eclipse.equinox.http.servlet.internal.servlet.RequestDispatcherAdaptor.include(RequestDispatcherAdaptor.java:48) ~[?:?]
	at com.liferay.portlet.internal.PortletRequestDispatcherImpl.dispatch(PortletRequestDispatcherImpl.java:291) ~[portal-impl.jar:?]
	... 181 more
Caused by: java.lang.IllegalArgumentException: Someone may be trying to circumvent the permission checker: {companyId=20097, name=com.liferay.document.library.kernel.model.DLFileEntryType, primKey=20100, scope=4}
	at com.liferay.portal.security.permission.AdvancedPermissionChecker._hasUserPermissionImpl(AdvancedPermissionChecker.java:1322) ~[portal-impl.jar:?]
	at com.liferay.portal.security.permission.AdvancedPermissionChecker._hasPermissionImpl(AdvancedPermissionChecker.java:1278) ~[portal-impl.jar:?]
	at com.liferay.portal.security.permission.AdvancedPermissionChecker.hasPermission(AdvancedPermissionChecker.java:243) ~[portal-impl.jar:?]
	at com.liferay.portal.security.permission.StagingPermissionChecker.hasPermission(StagingPermissionChecker.java:140) ~[portal-impl.jar:?]
	at com.liferay.portal.security.permission.StagingPermissionChecker.hasPermission(StagingPermissionChecker.java:156) ~[portal-impl.jar:?]
...
	at com.liferay.asset.internal.util.AssetHelperImpl.getAddPortletURL(AssetHelperImpl.java:150) ~[?:?]
	at com.liferay.asset.internal.util.AssetHelperImpl.getAssetPublisherAddItemHolders(AssetHelperImpl.java:432) ~[?:?]
	at com.liferay.product.navigation.control.menu.web.internal.display.context.AddContentPanelDisplayContext._getAddContentsURLs(AddContentPanelDisplayContext.java:314) ~[bundleFile:?]
	at com.liferay.product.navigation.control.menu.web.internal.display.context.AddContentPanelDisplayContext.lambda$getAddContentPanelData$0(AddContentPanelDisplayContext.java:98) ~[bundleFile:?]
	at com.liferay.portal.kernel.util.BaseMapWrapper.doPut(BaseMapWrapper.java:98) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.util.HashMapBuilder$HashMapWrapper.put(HashMapBuilder.java:135) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.util.HashMapBuilder.put(HashMapBuilder.java:58) ~[portal-kernel.jar:?]
	at com.liferay.product.navigation.control.menu.web.internal.display.context.AddContentPanelDisplayContext.getAddContentPanelData(AddContentPanelDisplayContext.java:94) ~[bundleFile:?]
	at org.apache.jsp.add_005fpanel_jsp._jspService(add_005fpanel_jsp:263) ~[?:?]
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:111) ~[jasper.jar:9.0.68]
...
	at org.eclipse.equinox.http.servlet.internal.servlet.RequestDispatcherAdaptor.include(RequestDispatcherAdaptor.java:48) ~[?:?]
	at com.liferay.portlet.internal.PortletRequestDispatcherImpl.dispatch(PortletRequestDispatcherImpl.java:291) ~[portal-impl.jar:?]
	... 181 more
Caused by: com.liferay.portal.kernel.exception.NoSuchResourcePermissionException: {companyId=20097, name=com.liferay.document.library.kernel.model.DLFileEntryType, primKey=20100, scope=4}
	at com.liferay.portal.service.impl.ResourcePermissionLocalServiceImpl.hasResourcePermission(ResourcePermissionLocalServiceImpl.java:1082) ~[portal-impl.jar:?]
	at sun.reflect.GeneratedMethodAccessor371.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_311]
...
	at com.liferay.depot.internal.security.permission.wrapper.DepotPermissionCheckerWrapper.lambda$hasPermission$3(DepotPermissionCheckerWrapper.java:110) ~[?:?]
	at com.liferay.depot.internal.security.permission.wrapper.DepotPermissionCheckerWrapper._hasPermission(DepotPermissionCheckerWrapper.java:209) ~[?:?]
	at com.liferay.depot.internal.security.permission.wrapper.DepotPermissionCheckerWrapper.hasPermission(DepotPermissionCheckerWrapper.java:108) ~[?:?]
	at com.liferay.portal.kernel.internal.security.permission.resource.DefaultModelResourcePermission._contains(DefaultModelResourcePermission.java:159) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.internal.security.permission.resource.DefaultModelResourcePermission.contains(DefaultModelResourcePermission.java:87) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.security.permission.resource.BaseModelResourcePermissionWrapper.contains(BaseModelResourcePermissionWrapper.java:59) ~[portal-kernel.jar:?]
	at com.liferay.document.library.web.internal.asset.model.DLFileEntryAssetRendererFactory.hasAddPermission(DLFileEntryAssetRendererFactory.java:241) ~[?:?]
	... 181 more

```

Reply:
Fixed. See https://github.com/liferay-lima/liferay-portal/pull/3924/commits/ece30a3501d8250b33efad4b6db6e8f48ae88596